Request to adding UML diagram

AlexRakitskiy opened the pull request:
- Class architecture UML  248b40f
- Added UML diagram classes  871e7da
- UML class architecture  8c375b8
- Fixed some UML classes  bd91419

pisarevski committed:
Re-created UML diagram

- Deleted old structure
- Added package trade with containing classes
- Added class Storage
- Added interface IStorageable
- Added interface IAttackable
- Added package ship with containing classes
- Added package parts with containing classes
- Added class Universe

AlexRakitskiy committed:
Added UML diagram classes

- Added ammunition package with containing classes
- Added locations package with containing classes
- Added events package with containing classes

pisarevski committed:
Fixed some UML diagram classes

- Fixed class Universe
- Fixed class Storage
- Fixed class Hull

AlexRakitskiy committed:
Fixed some UML classes

- Fixed ILandable interface
- Fixed Gun class

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,162 @@
+        -int spends;
+        +void startUniverse();
+        +void win();
+        +void lose();
```
`winState()`, `loseState()`

Alvahaze replied:
Fixed!

ViRGiL175 replied:
It is not fixed in last commit (49337ec9bd6e7fe8582bf00650133d44e242ae2b). @Alvahaze

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,162 @@
 
-            +void trade(Trader trader);
+    class Storage {
+        -HashMap<String, IStorageable> storage
```
storage?

Alvahaze replied:
Fixed!

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,162 @@
-            +void trade(Trader trader);
+    class Storage {
+        -HashMap<String, IStorageable> storage
+        +void add(String itemName);
```
Add by `itemName`?

Alvahaze replied:
Fixed!

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,162 @@
+    }
 
-            +void trade(Trader trader);
+    class Storage {
```
`get()` method?

Alvahaze replied:
Fixed!

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,162 @@
-    stuff <-[hidden]-- locations
+    interface IAttackable {
+        +void setDamage(int damage);
+        +int getHitPoints();
```
Where are hit points?

Alvahaze replied:
Fixed!

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,162 @@
+        Starship <|-- PirateStarship
 
-        class Planet {
+        abstract class Starship implements IAttackable {
```
Battle cycle.

AlexRakitskiy replied:
Fixed

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,162 @@
+
+    interface IStorageable {
+        +int getSize();
+        +String getName();
```
Where are names?

AlexRakitskiy replied:
Fixed

Alvahaze replied:
Fixed!

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,162 @@
     }
 
-    IAttackable <|.. Starship
+    package ammunition {
```
Why ammunition isn't `IStorageable`

AlexRakitskiy replied:
Fixed

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,162 @@
 
-    IReachable <|.. Starship
+        +abstract class Gun {
+            #int baseDamage;
```
Just `damage`?

AlexRakitskiy replied:
Fixed

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,162 @@
-    class Storage {
-        -HashMap<String, IStoreable> iStoreableHashMap;
+        +class Ammo {
+            -int baseDamage;
```
`damageIncreasing` maybe?

AlexRakitskiy replied:
Fixed

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,162 @@
+            +int getBaseDamage();
+        }
+
+        +enum AmmoType {
```
Shit

AlexRakitskiy replied:
Fixed

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,162 @@
-        ValuableItem <|-- MetalShell
-
-        class TurretMagazineCase {
+        +abstract class Event {
```
Constructor?

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,162 @@
+    }
 
-        }
+    interface ILandable {
```
Mindless?

AlexRakitskiy replied:
Fixed

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,162 @@
 
-            +void attack(IAttackable iAttackable);
+        abstract class Trader {
+            -int cash;
```
Where is storage?!

Alvahaze replied:
Fixed!

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,162 @@
-            -Railgun railgun;
-            -Turret rightTurret;
-            -Turret leftTurret;
+        class BaseTrader extends Trader {
```
Why 'base'?

Alvahaze replied:
Fixed!

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,162 @@
-            -Turret rightTurret;
-            -Turret leftTurret;
+        class BaseTrader extends Trader {
+            +void trade();
```
WTF?!!!

Alvahaze replied:
Fixed!

ViRGiL175 reviewed (changes requested):
- Fixes
- Commits messages (description details)

AlexRakitskiy committed:
Fixed UML diagram

- Added name and size fields for Gun and Magazine classes
- Gun and Magazine now implements IStorageable interface
- Renamed field in class Gun from baseDamage to damage
- Renamed field in class Ammo from baseDamage to damageIncreasing
- Removed field AmmoType from class Ammo and AmmoType enumerator
- Added new classes extends from Location
- Added new class Battle extends from Event

pisarevski committed:
UML architecture fixes

- Fixes in methods (class Universe)
- Changes in Storage class
- Changes in class Starship (added hitPoints)
- Changes in interface IStorageable
- Changes in class Trader

AlexRakitskiy committed:
Fixed UML diagram

- Fixed Battle class

AlexRakitskiy commented:
Changes done

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -5,9 +5,9 @@ package rape.brutal.gitstarship {
-        +win();
-        +lose();
+        +void startUniverse();
+        +void win();
```
It is not fixed. Why?

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -76,8 +76,8 @@ package rape.brutal.gitstarship {
-            +boolean putInStorage(IStoreable iStoreable);
-            +IStoreable getFromStorage(String iStoreableName);
+            +boolean putInStorage(IStoregable iStoregable);
+            +IStoregable getFromStorage(String iStoregableName);
```
"Storageable" or "storable" maybe?

http://www.dictionary.com/browse/storable

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,181 @@
+        -HashMap<String, IStorageable> items;
+        +void putItem(IStorageable iStorageable);
+        +void getItem(String itemName);
+        -int capacity;
```
Fields should be in top of class body.

Alvahaze replied:
Fixed

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,181 @@
 
-            -void flyTo(IReachable iReachable);
+        class AverageTrader extends Trader {
+            +void sell(int cash, String itemName);
```
Maybe these methods should return `boolean`?

Why void?

Alvahaze replied:
Fixed!

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,181 @@
-            +void trade(Trader trader);
+        class Mechanic extends Trader {
+            +void repair(int cash, String itemName);
+            +void upgrade(int cash, String itemName);
```
Maybe `boolean` too?

In case of lack of money etc?

Alvahaze replied:
Fixed!

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,181 @@
-            -Railgun railgun;
-            -Turret rightTurret;
-            -Turret leftTurret;
+        +abstract class Event {
```
Possible NPE (`NullPointerException`) for `universe`. Class _should_ have `universe` for callbacks.

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,181 @@
 
-        class PirateStarship {
-            -Turret rightTurret;
+        +class Battle {
```
Very stupid class.

Possible NPE for `gitStarship`, NPE for `universe` etc.

What is purpose of this class?

Alvahaze replied:
Fixed!

ViRGiL175 commented on src/main/resources/diagram_git_starship.puml:
```diff
@@ -2,132 +2,181 @@
-        abstract class ValuableItem {
-            -int value;
+        +class OpenSpace extends Location {
+            -ArrayList<Starship> ships;
```
Why there are no possibility to find storage in open space?

It can be a pieces of blown Starship or life-saving capsule etc.

Alvahaze replied:
Fixed!

ViRGiL175 reviewed (changes requested):
- Where is loot and different tradable stuff?
- Where are constructors? Especially in `Event`, `Starship` classes. It is a possible NPE in future.
- Confusing `Battle` class

Nice progress anyway.
